src/syntax-highlighting.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  buildRegExp,
+  capture,
+  charClass,
+  choiceOf,
+  endOfString,
+  notWhitespace,
+  startOfString,
+  whitespace,
+  word,
+  zeroOrMore,
+} from 'ts-regex-builder';
+
+const BACKTICKS = '```';
+
+export const codeBlockRegExp = buildRegExp(
+  [capture([BACKTICKS, zeroOrMore(charClass(whitespace, notWhitespace)), BACKTICKS])],
+  {
+    global: true,
+    ignoreCase: true,
+  }
+);
```

Comment:
`whitespace` + `notWhitespace` is essentially a full `any`. What we want here is non-greedy quantifier (`zeroOrMore` or `oneOrMore`) that would match the nearest BACKTICKS.

This will avoid issue where are are two code blocks and a greedy quantifier would match the opening BACKTICKS from the first one and closing BACKTICKS from the second one, considering everything inside to be a single code block.


```suggestion
export const codeBlockRegExp = buildRegExp(
  [
    capture([
      BACKTICKS,
      oneOrMore(any, { greedy: false }),
      BACKTICKS,
    ]),
  ],
  {
    global: true,
    ignoreCase: true,
  }
```